core/store/src/migrations/v8_to_v9.rs

```diff
@@ -27,14 +27,18 @@ where
         store_update.update_refcount(column, key.as_ref(), &value, 1);
         batch_size += key.as_ref().len() + value.len() + 8;
         if batch_size > batch_size_limit {
-            store_update.commit().expect(&format!("Failed during recomputing column {:?}", column));
+            store_update
+                .commit()
+                .unwrap_or_else(|_| panic!("Failed during recomputing column {:?}", column));
```

Comment:
so unwrap_or_else with panic is better than expect ? (@matklad )

Reply:
In this case, yes. The difference is that 

```rust
.expect(&format!("Failed during recomputing column {:?}", column));
```

creates a heap-allocated string even on the happy path. In contrast, 

```
.unwrap_or_else(|_| panic!("Failed during recomputing column {:?}", column));
```

only allocates on a sad path. 

THat is, `.expect("str literal")` is good, `.expect(&format!("string formatting at {}", "run time"))` is needlessly slow.